src/MPoly.jl

```diff
@@ -445,6 +448,13 @@ function content(a::MPolyRingElem{T}) where T <: RingElement
    return z
 end
 
+
+function is_nilpotent(f::T) where {T<:MPolyRingElem}
+  is_domain_type(elem_type(coefficient_ring(f))) && return is_zero(f)
```

Comment:
```suggestion
function is_nilpotent(f::MPolyRingElem{T}) where {T}
  is_domain_type(T) && return is_zero(f)
```
would be IMHO more idiomatic (and simply much shorter to read).

The same could of course be used in `is_unit` above and elsewhere.

Reply:
OK, I'll try.  I have found `is_domain_type` to be quite obstreperous, which is why I have used work-arounds.  Maybe I should update...

Reply:
Could you clarify what you found problematic about `is_domain_type`? What are you "working around"?

Reply:
Seems to work now -- I no longer recall what problem I had.  Sorry!